riscv_isac/InstructionObject.py

```diff
@@ -316,6 +328,129 @@ def evaluate_instr_var(self, instr_var_name, *args):
 
         return None
 
+    def ptw_update(self,instr_vars):
+        '''
+        This function calculates the virtual, physical address of instruction,
+        data and also the page table walk addresses 
+        for the data.
+
+        :param instr_vars : dictionary to be populated by the evaluated address variables
+                            in case when the virtual memory is mentioned under the config
+                            label.
+        '''
+        match = ['VM']
+        if instr_vars['xlen'] == 32:
+            if ((instr_vars['satp']) >> 31) == 1:
+                match.append('SV32')
+            else:
+                match.append(0)
+        elif instr_vars['xlen'] == 64:
+            if ((instr_vars['satp']) >> 60) == 8:
+                match.append('SV39')
+            elif ((instr_vars['satp']) >> 60) == 9:
+                match.append('SV48')
+            elif ((instr_vars['satp']) >> 60) == 10:
+                match.append('SV57')
+            else:
+                match.append(0)
+        if match[0] == 'VM' and match[1] in ['SV32', 'SV39', 'SV48', 'SV57']:
+            instr_vars['depa'] = self.vm_addr_dict['depa']
+            instr_vars['ieva'] = self.vm_addr_dict['ieva']
+            instr_vars['iepa'] = self.vm_addr_dict['iepa']
+            instr_vars['ieva_align'] = self.vm_addr_dict['ieva_align']
+            instr_vars['iepa_align'] = self.vm_addr_dict['iepa_align']
+            instr_vars['depa_align'] = self.vm_addr_dict['depa_align']
+            format_max_len_mapping = {'SV32': 2, 'SV39': 3, 'SV48': 4, 'SV57': 5}
+            max_len = format_max_len_mapping.get(match[1],0)
+            size = len(self.vm_addr_dict['dptw_list'])
+            instr_vars['len_dptw'] = size
+            remain = max_len
+            length = max_len - size - 1
+
+            if (max_len >= size):
+                for i in range(size):
+                    instr_vars[f'dptw{max_len-1}a'] = int(self.vm_addr_dict['dptw_list'][i][0], 16)
+                    instr_vars[f'dptw{max_len-1}cont'] = int(self.vm_addr_dict['dptw_list'][i][1], 16)
+                    max_len = max_len -1
+                for i in range(length, -1, -1):
+                    instr_vars[f'dptw{i}cont'] = None
+                    instr_vars[f'dptw{i}a'] = None
+                for i in range(remain, 5):
+                    instr_vars[f'dptw{i}cont'] = None
+                    instr_vars[f'dptw{i}a'] = None
+            else:
+                raise Exception("Implemented Virtual Memory in the Test and the one given in the coverpoints is not same.")
```

Comment:
I think this particular error message is misleading. If I understand correctly, this scenario should never occur as the SATP is tracked from the logs and should match the PT walk scheme being used.

Reply:
Yes, thank you for the correction. I have pushed the corrected commit to this PR.